### PR TITLE
[New] Style features with custom dictionary

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Style features with custom dictionary` in `Visualization` category.
URL to README: [ReadME](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-StyleFeaturesWithCustomDictionary/Shared/Samples/Style%20features%20with%20custom%20dictionary)

## Linked Issue(s)

- `swift/issues/4074`

## How To Test

- Toggle between "Style File" and "Web Style" to see the dictionary symbols on the map change.

## Screenshots

style-features-with-custom-dictionary screenshot

### CODE REVIEW

**Comment by CalebRas** [comment]
Removed Ting as a reviewer.

File: Samples.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -151,6 +151,7 @@
 		D722BD222A420DAD002C2087 /* ShowExtrudedFeaturesView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D722BD212A420DAD002C2087 /* ShowExtrudedFeaturesView.swift */; };
 		D722BD232A420DEC002C2087 /* ShowExtrudedFeaturesView.swift in Copy Source Code Files */ = {isa = PBXBuildFile; fileRef = D722BD212A420DAD002C2087 /* ShowExtrudedFeaturesView.swift */; };
 		D734FA0C2A183A5B00246D7E /* SetMaxExtentView.swift in Sources */ = {isa = PBXBuildFile; fileRef = D734FA092A183A5B00246D7E /* SetMaxExtentView.swift */; };
+		D73F8CF42AB1089900CD39DA /* Restaurant.stylx in Resources */ = {isa = PBXBuildFile; fileRef = D73F8CF32AB1089900CD39DA /* Restaurant.stylx */; settings = {ASSET_TAGS = (AddCustomDynamicEntityDataSource, StyleFeaturesWithCustomDictionary, ); }; };
```
Comment (zkline101) [inline Samples.xcodeproj/project.pbxproj:154]: You can remove the `AddCustomDynamicEntityDataSource` asset tag.

File: Shared/Samples/Style features with custom dictionary/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+## How to use the sample
+
+Toggle between the dictionary symbols from the web style and style file. Pan and zoom around the map to see the symbology from the chosen dictionary symbol style. The web style and style file are slightly different to each other to give a visual indication of the switch between the two.
+
+## How it works
+
+1. Create a `PortalItem`, referring to a `Portal` and the item ID of the web style.
+2. Based on the style selected:
+*  If the web style toggle has been selected, create a new `DictionarySymbolStyle` from the portal item using `DictionarySymbolStyle(portalItem:)`, and load it.
+*  If the file style toggle has been selected, create a new `DictionarySymbolStyle` using `DictionarySymbolStyle.createFromFile(url:)`.
```
Comment (zkline101) [inline Shared/Samples/Style features with custom dictionary/README.md:20]: In the Swift API, it is not `DictionarySymbolStyle.createFromFile(url:)`, it is [DictionarySymbolStyle(url:)](https://developers.arcgis.com/swift/api-reference/documentation/arcgis/dictionarysymbolstyle/init(url:)).

File: Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        
+        init() {
+            // Add the feature layer with the dictionary renderer to the map.
+            restaurantFeatureLayer.renderer = dictionaryRendererFromStyleFile
+            map.addOperationalLayer(restaurantFeatureLayer)
+        }
+    }
+    
+    /// The custom dictionary symbol styles to choose from.
+    enum CustomDictionaryStyle: CaseIterable {
```
Comment (zkline101) [inline Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift:107]:
```suggestion
    private enum CustomDictionaryStyle: CaseIterable {
```

File: Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            return FeatureLayer(featureTable: restaurantFeatureTable)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style dictionary file (.stylx) on local disk.
+        let dictionaryRendererFromStyleFile: DictionaryRenderer = {
+            // The URL to the symbol style dictionary file.
+            let restaurantStyleURL = URL.restaurants
+            
+            // Create the dictionary symbol style from the style file URL.
+            let restaurantStyle = DictionarySymbolStyle(url: restaurantStyleURL)
```
Comment (zkline101) [inline Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift:72]:
```suggestion
            // Create the dictionary symbol style from the style file URL.
            let restaurantStyle = DictionarySymbolStyle(url: .restaurants)
```

File: Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            // Create a dictionary symbol style from the web style portal item.
+            let restaurantStyle = DictionarySymbolStyle(portalItem: restaurantPortalItem)
+            
+            // Create a dictionary renderer from the dictionary symbol style.
+            // Map the input fields in the feature layer to the dictionary symbol
+            // style's expected fields for symbols.
+            return DictionaryRenderer(
+                dictionarySymbolStyle: restaurantStyle,
+                symbologyFieldOverrides: ["healthgrade": "Inspection"],
+                textFieldOverrides: [:]
```
Comment (zkline101) [inline Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift:95]: This parameter has a default value:
```suggestion
                symbologyFieldOverrides: ["healthgrade": "Inspection"]
```

File: Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+            let restaurantStyle = DictionarySymbolStyle(url: restaurantStyleURL)
+            
+            // Create the dictionary renderer from the dictionary symbol style.
+            return DictionaryRenderer(dictionarySymbolStyle: restaurantStyle)
+        }()
+        
+        /// A dictionary renderer created from a custom symbol style hosted on ArcGIS Online.
+        let dictionaryRendererFromWebStyle: DictionaryRenderer = {
+            // The create a portal item with the id of the online web style.
+            let restaurantPortalItem = PortalItem(
```
Comment (zkline101) [inline Shared/Samples/Style features with custom dictionary/StyleFeaturesWithCustomDictionaryView.swift:81]: Repeating type information in the variable isn't that helpful:
```suggestion
            let restaurant = PortalItem(
```

**Comment by zkline101** [comment]
@CalebRas Is this open for review? If so, please re-request reviews.

**Review by zkline101 — DISMISSED** [review]

**Review by zkline101 — APPROVED** [review]

**Review by pgruenler — APPROVED** [review]

**Pull request merged**